extensions/basic-ui/src/AppBar.tsx

```diff
@@ -17,10 +18,23 @@ interface AppBarProps {
   title?: React.ReactNode;
   appendLeft?: () => React.ReactNode;
   appendRight?: () => React.ReactNode;
+  backButton?:
+    | {
+        renderIcon?: () => React.ReactNode;
+        onClick?: () => void;
+      }
+    | {
+        render?: () => React.ReactNode;
+      };
+  closeButton?:
```

Comment:
```suggestion
  | {
       onClick: () => void;
    }
  | {
        renderIcon?: () => React.ReactNode;
        onClick: () => void;
      }
    | {
        render?: () => React.ReactNode;
      };
```

이렇게 optional이 아닌 경우에는 onClick을 강제하는 디자인은 어떨까요?

1. renderIcon을 사용하지 않을 경우에 onClick을 mandatory로 강제시킴. (기본동작이 제공될 것이라는 오해를 줄여줌)
2. renderIcon만 하면 기본 onClick event가 있을 것 같은 오해를 줄여줌.
3. renderIcon과 render의 구분이 명확해짐. render는 onClick이 없으므로 완전한 동작구성을 제공해야 될 것이라고 추측하기 쉬움

Reply:
```
| {
	       onClick: () => void;
	    }
	  | {
	        renderIcon?: () => React.ReactNode;
	        onClick: () => void;
	      }
```

요 두개는 같은 표현이라서 하나로 합칠게요 ㅎㅎ